app/screens/buyers/BuyingItemList.java

```diff
@@ -1,44 +1,90 @@
 package icebase.app.screens.buyers;
 
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Scanner;
 
 import icebase.app.App;
 import icebase.app.MenuTitle;
+import icebase.app.api.API;
+import icebase.app.enums.CATEGORY_ENUM;
+import icebase.app.exceptions.CategoryDoesNotExistException;
 import icebase.app.screens.Screen;
-import java.util.Scanner;
+import icebase.app.types.Store;
+import icebase.app.types.items.Item;
+import icebase.icebase.Auth;
+import icebase.icebase.User;
 
 public class BuyingItemList implements Screen {
 
-    public void display() {
-        Scanner sc = App.sc;
-        int choice;
-        String storeName = "STORE X"; // stand-in value, should be set from constructor
-        String categoryName = "CATEGORY X"; // stand-in value, should be set from constructor
-        String[] itemList = { "Item A", "Item B", "Item C", "Return" }; // fetch from icebase
-        int range = itemList.length;
+    private String storeName;
+    private String categoryName;
+    private List<Item> itemList;
+    private ArrayList<String> itemNames;
+    private Scanner sc = App.sc;
+    private Auth auth = Auth.getAuth();
+    private User user = auth.getUser();
+
+    public BuyingItemList(Store store, CATEGORY_ENUM category) {
+        this.storeName = store.getName();
+        this.categoryName = category.value;
+        try {
+            this.itemList = API.getStoreItems(store, category);
+            for (Item item : itemList) {
+                itemNames.add(item.getName());
+            }
+            itemNames.add("return");
+
+        } catch (IOException io) {
+            io.printStackTrace();
+        } catch (CategoryDoesNotExistException cdne) {
+            cdne.printStackTrace();
```

Comment:
stack trace ang i print or ang medsage lang?

Reply:
which do you think is better man?

Reply:
user experience kay daoat ang message lang, i created custom messages sa different exceptions so yun lang i print

i preserve lang ang trackTrace sa mga unexpected like ioexception and other wala na expect

Reply:
gotchu